### PR TITLE
Update apis for imporved rendering for cms

### PR DESCRIPTION
**Pull request by weastel**
Signed-off-by: Drumil Patel <drumilpatel720@gmail.com>

### CODE REVIEW

**Comment by weastel** [comment]
@algomaster99 Please review this

File: serializers.py [hunk 1]
```diff
@@ -58,7 +64,28 @@ def get_is_completed(self, instance):
...
+
+            for sem in SEMESTER_TYPES:
+                if sem[0] == instance.semester:
+                    return sem[1]
+            
+            return ""
+
     return Serializer
 
+
```
Comment (algomaster99) [inline serializers.py:89]: Why is this extra new line added?

File: serializers.py [hunk 2]
```diff
@@ -58,7 +64,28 @@ def get_is_completed(self, instance):
...
+            return "A" if instance.has_already_ended else "O"
+
+        def get_semester_name(self, instance):
+            """
+            Returns semester name from semester annotations. i.e Spring for s.
+            :return: semester name from semester annotations
+            """
+
+            for sem in SEMESTER_TYPES:
```
Comment (algomaster99) [inline serializers.py:81]: Django has a function called `get_semester_display()` (get_choice field name_display) which can fetch you the display text of the choices.

File: serializers.py [hunk 3]
```diff
@@ -58,7 +64,28 @@ def get_is_completed(self, instance):
 
             return instance.has_already_ended
 
+        def get_status_type(self, instance):
```
Comment (algomaster99) [inline serializers.py:67]: I now feel that this should be a part of CMS as this is not related to FacApp. You can simply check for `isCompleted ` there.

File: serializers.py (hunk 2 above)
Comment (algomaster99) [reply serializers.py:81]: I would also prefer if this function can be written in `cms-omniport-api` instead.

File: serializers.py [hunk 4]
```diff
@@ -1,5 +1,7 @@
 import swapper
 
+from faculty_biodata.constants.semesters import SEMESTER_TYPES
```
Comment (algomaster99) [inline serializers.py:3]: Not required as explained below.

**Comment by weastel** [comment]
Closing this PR in favour of [PR](https://github.com/IMGIITRoorkee/cms-omniport-api/pull/5)
